crates/sui-rpc-benchmark/src/direct/query_executor.rs

```diff
@@ -95,7 +95,10 @@ impl QueryExecutor {
         );
 
         let start = Instant::now();
-        let deadline = start + self.config.duration;
+        let deadline = match self.config.duration {
```

Comment:
Not strictly related to this PR, but some clean-ups in the call to `try_for_each_spawned`:

- Could you make `deadline` optional on `worker_task` as well instead of picking some far off time?
- The step where you take each query and then wrap it up in a tuple of other values seems unnecessary. I think the whole rest of the function could look like so:

```rust
futures::stream::iter(queries)
    .try_for_each_spawned(concurrency, |query| {
        QueryExecutor::worker_task(
            self.pool.clone(),
            vec![query],
            self.metrics.clone(),
            deadline,
        )  
    })
    .await?;

Ok(self.metrics.generate_report())
```